Starforge/Editor/ToolManager.cs

```diff
@@ -35,26 +33,51 @@ public static class ToolManager {
         public static int FGTileset;
 
         static ToolManager() {
-            Tools = new Dictionary<ToolType, Tool>()
-            {
-                [ToolType.TileBrush] = new TileBrushTool(),
-                [ToolType.TileRectangle] = new TileRectangleTool(),
-                [ToolType.Entity] = new EntityTool()
-            };
-
-            SelectedTool = Tools[ToolType.TileBrush];
+            Tools = new Dictionary<string, Tool>();
+
             SelectedLayer = ToolLayer.Foreground;
 
             BGTileset = 0;
             FGTileset = 0;
         }
 
         public static void Render() {
+            if (SelectedTool == null)
+                SelectedTool = Tools["TileBrush"];
             SelectedTool.Render();
         }
 
         public static void Update() {
+            if (SelectedTool == null)
+                SelectedTool = Tools["TileBrush"];
             SelectedTool.Update();
         }
+
+        public static void Register(Type type) {
+            try {
+                if (!type.IsSubclassOf(typeof(Tool))) return;
```

Comment:
Can't we just use a Tool Parameter instead and use polymorphy instead of this? Should be the same thing, just without that extra check and safer in general

Reply:
Assuming you meant using `Register(Tool tool)`, that requires an instance of that Tool, which we don't have yet (it gets created within this method). Registering using a Type is also what the EntityManager does.

Reply:
Oh right, i misunderstood what this is doing, sorry then